python/ruff/__main__.py

```diff
@@ -3,6 +3,23 @@
 import sysconfig
 from pathlib import Path
 
+RUFF_PATHS = [
+    Path(sysconfig.get_config_var("userbase")) / "bin" / "ruff",
+    Path(sysconfig.get_path("scripts")) / "ruff",
+]
+
+
+def find_ruff_bin() -> Path:
+    """Return the ruff binary path."""
+    for ruff_path in RUFF_PATHS:
+        if ruff_path.is_file():
+            return ruff_path
+    raise FileNotFoundError(ruff_path)
+
+
 if __name__ == "__main__":
-    ruff = Path(sysconfig.get_path("scripts")) / "ruff"
+    try:
+        ruff = find_ruff_bin()
+    except FileNotFoundError as e:
+        raise FileNotFoundError(e) from e
```

Comment:
This ends up raising `FileNotFoundError(FileNotFoundError(ruff_path))`. Why catch and rewrap the exception at all?